xls/examples/parameterized/BUILD

```diff
@@ -0,0 +1,105 @@
+load("//xls/build_rules:xls_utilities.bzl", "proto_to_dslx")
+load("@bazel_skylib//rules:common_settings.bzl", "string_flag")
+load(
+    "//xls/build_rules:xls_build_defs.bzl",
+    "xls_dslx_ir",
+    "xls_dslx_library",
+    "xls_ir_opt_ir",
+    "xls_ir_verilog",
+)
+
+proto_library(
+    name = "proto_schema_lib",
+    srcs = [":param.proto"],
+)
+
+# Parameters for the narrow variant of the design
+proto_to_dslx(
+    name = "parameters_narrow",
+    proto_schema = ":proto_schema_lib",
+    proto_name = "xls.DesignVariants",
+    textproto = ":narrow.textproto",
+)
+
+xls_dslx_library(
+    name = "parameters_narrow_lib",
+    srcs = [":parameters_narrow"],
+)
+
+# Parameters for the wide variant of the design
+proto_to_dslx(
+    name = "parameters_wide",
+    proto_schema = ":proto_schema_lib",
+    proto_name = "xls.DesignVariants",
+    textproto = ":wide.textproto",
+)
+
+xls_dslx_library(
+    name = "parameters_wide_lib",
+    srcs = [":parameters_wide"],
+)
+
+# Design implementation rules - narrow variant
+xls_dslx_library(
+    name = "design_narrow_lib",
+    srcs = ["design_narrow.x"],
+    deps = [":parameters_narrow_lib"],
+)
+
+xls_dslx_ir(
+    name = "design_narrow_ir",
+    dslx_top = "top",
+    ir_file = "desing_narrow.ir",
+    library = "design_narrow_lib",
+)
+
+xls_ir_opt_ir(
+    name = "design_narrow_opt_ir",
+    src = "desing_narrow.ir",
+    top = "__design_narrow__top",
+)
+
+xls_ir_verilog(
+    name = "design_narrow_verilog",
+    src = ":design_narrow_opt_ir.opt.ir",
+    codegen_args = {
+        "module_name": "top",
+        "delay_model": "unit",
+        "pipeline_stages": "1",
+        "use_system_verilog": "false",
+    },
+    verilog_file = "design_narrow.v",
+)
+
+# Design implementation rules - wide variant
+xls_dslx_library(
+    name = "design_wide_lib",
+    srcs = ["design_wide.x"],
+    deps = [":parameters_wide_lib"],
```

Comment:
Attempting to put results of `proto_to_dslx` target straight in `xls_dslx_library` in `src` attribute beside `design.x` results in [error](https://github.com/google/xls/blob/919e7d8e61465973259b3287d703bcdc3ff2d12f/xls/build_rules/xls_ir_rules.bzl#L596):
```
Error in fail: A single source file must be specified.
```

If I understand correctly, this would probably require keeping rules for building IRs from each source DSLX file and then merging those in codegen.

Reply:
This doesn't seem great for usability but no need for it to block landing this example.